### PR TITLE
Modify rule S3649(Python): Update text to Education Framework structure (APPSEC-189)

### PR DESCRIPTION
**Pull request by pedro-oliveira-sonarsource**

### CODE REVIEW

**Comment by pedro-oliveira-sonarsource** [comment]
secure-by-design.adoc is being done in this [PR](https://github.com/SonarSource/rspec/pull/1355).

**Comment by pedro-oliveira-sonarsource** [comment]
A hardening [ticket](https://sonarsource.atlassian.net/browse/APPSEC-215) to add Django content was created.

File: rules/S3649/python/how-to-fix-it/sqlite3.adoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+The following code is an example of an overly simple data retrieval function.
+It is vulnerable to SQL injection because user-controlled data is inserted
+directly into a query string: The application assumes that incoming data
+always has a specific range of characters and ignores that some characters may
+change the query logic to a malicious one.
+
+In this particular case, the query can be exploited with the following string: 
+
+----
+' OR 1='1
```
Comment (gaetan-ferry-sonarsource) [inline rules/S3649/python/how-to-fix-it/sqlite3.adoc:12]: This payload does not work with SQLite that is strongly typed.
```
sqlite> select 1 where 1='1';
sqlite>
```
```suggestion
' OR '1'='1
```

File: rules/S3649/python/how-to-fix-it/sqlite3.adoc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+----
+
+Using the UNION clause, an attacker would be able to perform queries against
```
Comment (gaetan-ferry-sonarsource) [inline rules/S3649/python/how-to-fix-it/sqlite3.adoc:15]:
```suggestion
Using the UNION clause, an attacker would also be able to perform queries against
```

**Review by gaetan-ferry-sonarsource — CHANGES_REQUESTED** [review]
LGTM in general, with a few suggestions and fixes.

Two other things not in the comments:

1. The rule.adoc does not include the two how to fix files. It raises a testing error.
2. I think there is something missing regarding the prepared statement fix. It is not possible to parameterize the SQL keywords or table names. I think we could add a paragraph in the prepared statement how to fix with this and propose solutions. A pitfall could probably work too.

**Comment by pedro-oliveira-sonarsource** [comment]
>     2. I think there is something missing regarding the prepared statement fix. It is not possible to parameterize the SQL keywords or table names. I think we could add a paragraph in the prepared statement how to fix with this and propose solutions. A pitfall could probably work too.

Think creating a hardening ticket for this would be appropriate.

**Review by gaetan-ferry-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**